services/tables/src/main/java/com/linkedin/openhouse/tables/config/MainApplicationConfig.java

```diff
@@ -74,6 +81,32 @@ public UserTableApi provideApiInstance() {
     return new UserTableApi(apiClient);
   }
 
+  @Bean
+  public WebMvcTagsContributor clientIdTagContributor() {
+    return new WebMvcTagsContributor() {
+      @Override
+      public Iterable<Tag> getTags(
+          HttpServletRequest request,
+          HttpServletResponse response,
+          Object handler,
+          Throwable exception) {
+        String clientName = request.getHeader(HTTPHEADER_CLIENT_NAME);
```

Comment:
do we need to deal with null value here?

Reply:
https://docs.oracle.com/cd/E17802_01/products/products/servlet/2.5/docs/servlet-2_5-mr2/javax/servlet/http/HttpServletRequest.html

documents that getHeader will return null if not found. as well, there isn't an exception in the interface to be thrown if the value isn't found. 

then the value is used in "List<String>.contains()" later which will attempt to find the null value  in the list which won't exist. so this code as is should handle null. 

this case is under test too, `testMetricsWithClientNameHeaderNotPresentDefaultValue`